sys/cheri/cheric.h

```diff
@@ -124,7 +120,7 @@ cheri_is_address_inbounds(const void * __capability cap, vaddr_t addr)
  * cheri_getpcc() for now.
  */
 static __inline void * __capability
-cheri_codeptr(const void *ptr, size_t len)
+cheri_codeptr(void *ptr, size_t len)
```

Comment:
It would be nice to keep the `const` and const-cast inside instead, like the string functions.

Reply:
I've made it type preserving now. This shouldn't really matter since it's only used in libcheri.